reproducibility_project/src/engines/mcccs/analysis.py

```diff
@@ -0,0 +1,113 @@
+    data_path = "analysis_data"
+    if os.path.exists(data_path):
+        shutil.rmtree(data_path)
+    os.makedirs(data_path)
+    # delete the folder manually
+
+    os.chdir(data_path)
+
+    project = signac.get_project()
+
+    for (molecule, ensemble, temperature, pressure), group in project.groupby(
+        (
+            "molecule",
+            "ensemble",
+            "temperature",
+            "pressure",
+        )
+    ):
+        print(molecule, ensemble, temperature, pressure)
+        if not os.path.isdir(
+            "{}_{}_{}K_{}kPa".format(molecule, ensemble, temperature, pressure)
+        ):
+
+            os.makedirs(
+                "{}_{}_{}K_{}kPa".format(
+                    molecule, ensemble, temperature, pressure
+                )
+            )
+        os.chdir(
+            "{}_{}_{}K_{}kPa".format(molecule, ensemble, temperature, pressure)
+        )
+
+        base_dir = os.getcwd()
+        if ensemble == "NPT" and molecule == "methaneUA":
+
+            density_list = []
+            for job in group:
+                os.chdir(job.ws)
+                #        print(job)
+                prod_run_files = sorted(glob("run*prod*"))
+                if len(prod_run_files) < 4:
+                    print(
+                        "warning, only {}  prod cycles complete for {} {} {} {} {}".format(
+                            len(prod_run_files),
+                            job,
+                            molecule,
+                            ensemble,
+                            temperature,
+                            pressure,
+                        )
+                    )
+                density_list.append(avg_one_seed_density(prod_run_files))
+
+            #    print(density_list)
+            filtered_density_list = list(filter(None, density_list))
+            output_string = "The average density is {} g/ml with SEM {} from {} samples".format(
+                np.mean(filtered_density_list),
+                np.std(filtered_density_list)
+                / np.sqrt(len(filtered_density_list)),
+                len(filtered_density_list),
+            )
+            print(output_string)
+            os.chdir(base_dir)
+            text_file = open("density_results.txt", "w")
+            text_file.write(output_string)
+            text_file.close()
+
+            # comb_traj = md.load_hdf5("comb_traj.h5")
+            # print(comb_traj)
+        elif ensemble == "GEMC-NVT" and molecule == "methaneUA":
+            for job in group:
+                print(job)
+
+        os.chdir("..")
+
+
+def avg_one_seed_density(prod_run_files):
+    """For a one particular seed, read all the prod run files and provide the average density (g/ml) for one seed."""
+    if len(prod_run_files) == 0:
+        return None
+    os.system(
+        "grep 'specific density                        ' run.prod* | awk '{print $6'} > temp_density.txt"
```

Comment:
Piggybacking off of this, you might be able to just load this into a pandas dataframe using [read_csv](https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html)
```py3
import pathlib
import pandas as pd

file_path = pathlib.Path(job.workspace())
prod_files = file_path.glob("run.prod*")
my_df = pd.concat([pd.read_csv(f, header=0, delim_whitespace=True) for f in prod_files])

vals = my_df["specific_density"].to_numpy()
```

You might have to change some of the logic I put above, but I would play around with that and see if it works.

Reply:
Thanks for the suggestion @rwsmith7531 . The code is giving me same results irrespective of the position of the single quote, but it looks better to have it after the curly brace.
@justinGilmer 
I am using `grep` and `awk` because my run files are text heavy. `pandas` is unable to read such a file. I am appending one example output file at the end of this comment. If you think a Python based function can be used more efficiently, please let me know. 

`
Program started at 2021-09-14 18:24:46.254
Number of processors: 1
Threads per processor: 1

MCCCS topmon (branch: ABE)
Commit hash: f9f01f4
Build on host: metropolis2.chem.umn.edu
Preprocessor definitions: __DOUBLE_PRECISION__;__USEOWN__
Using Intel compiler: /share/apps/intel/composer_xe_2013.1.117/bin/intel64/ifort

 run_num: 1
 suffix: a
 L_movie_xyz:  T
 L_movie_pdb:  T

***** PROGRAM  =  THE MAGIC BLACK BOX *****
Isobaric-isothermal ensemble
Using periodic boundaries in x y z
3-dimensional periodic box
Particle coordinates are folded into central box
CBMC simultaneously grows all beads conected to the same bead
   with bond lengths/angles generated from Gaussian distribution
Program will call explct() for explicit-hydrogen models
Dual Cutoff Configurational-bias Monte Carlo
Coulombic interactions will be included in the Rosenbluth weights for CBMC growth
Coulombic inter- and intra-molecular interactions will be calculated
   using (neutral-)group-based cutoff
Lorentz-Berthelot combining rules apply
 L_spline:  F
 L_linear:  F
*******************************************

NAMELIST MC_SHARED
------------------------------------------
Random number seed: 12
number of boxes in the system: 1
number of molecule types: 1
number of chains: 400
number of cycles: 5000
 iratio: 500
minimum cutoff (rmin):   1.000 [Ang]
 softcut:  100.0000000
Write checkpoint file every 1800 seconds, and keep the last 1 copies
 linit:  F
 lreadq:  F

NAMELIST ANALYSIS
------------------------------------------
 iprint: 50
 imv: 1000
 iblock: 1000
 iratp: 500
 idiele: 100000
 iheatcapacity: 1000000
 ianalyze: 100000
 nbin: 250
 lrdf:  T
 lintra:  F
 lstretch:  F
 lgvst:  F
 lbend:  F
 lete:  F
 lrhoz:  F
 bin_width:  0.200000000E-01
 lucall:  F

SECTION SIMULATION_BOX
------------------------------------------
Box 1:   42.170 x   42.170 x   42.170
   rcut: 14.000 [Ang], kalp:  0.250
   neighbor list cutoff (rcutnn):  0.000
   number of dimensions that are isotropic: 3
   lsolid:  F, lrect:  F, lideal:  F, ltwice:  F
   temperature: 1000.000 [K]
   external pressure:   2.26000000     [MPa]
   initial number of chains of each type:   400
   Ghost particles: 0
   initial number of chains in x, y and z directions: 10 x 10 x 2
   initial rotational displacement: 10, inimix: 0, zshift:  10.0, dshift:   2.000

NUMBER OF MOLECULES OF EACH TYPE
 number of chains of each type:   400

SECTION MOLECULE_TYPE
------------------------------------------
molecule type:  1 L1
   number of units: 6
   number of units for CBMC growth: 6
   number of carbons for EH alkane: 6
   maximum number of units for CBMC: 6
   maximum number of interior segments for SAFE-CBMC regrowth: 6
   number of atoms in a ring (if lring=.true.): 6
   iurot: 0, isolute: 100000, lelect:  F, lring:  F, lrigid:  F, lbranch:  F, lsetup:  F, lq14scale:  F, qscale: 1.0

   bead 1: bead type 1002 [_CH], charge leader 1
      bonded to bead 2, type 2001, bond length:  1.40000, k/2:   236011.450
      bonded to bead 6, type 2001, bond length:  1.40000, k/2:   236011.450
      bending interaction through 2 with bead 3, bending type: 3001, bending angle:  120.000, k/2:   31702.5000
      bending interaction through 6 with bead 5, bending type: 3001, bending angle:  120.000, k/2:   31702.5000
      torsional interaction through 2 and 3 with bead 4, torsional type: 4001
      torsional interaction through 6 and 5 with bead 4, torsional type: 4001

   bead 2: bead type 1002 [_CH], charge leader 2
      bonded to bead 1, type 2001, bond length:  1.40000, k/2:   236011.450
      bonded to bead 3, type 2001, bond length:  1.40000, k/2:   236011.450
      bending interaction through 3 with bead 4, bending type: 3001, bending angle:  120.000, k/2:   31702.5000
      bending interaction through 1 with bead 6, bending type: 3001, bending angle:  120.000, k/2:   31702.5000
      torsional interaction through 3 and 4 with bead 5, torsional type: 4001
      torsional interaction through 1 and 6 with bead 5, torsional type: 4001

   bead 3: bead type 1002 [_CH], charge leader 3
      bonded to bead 2, type 2001, bond length:  1.40000, k/2:   236011.450
      bonded to bead 4, type 2001, bond length:  1.40000, k/2:   236011.450
      bending interaction through 4 with bead 5, bending type: 3001, bending angle:  120.000, k/2:   31702.5000
      bending interaction through 2 with bead 1, bending type: 3001, bending angle:  120.000, k/2:   31702.5000
      torsional interaction through 4 and 5 with bead 6, torsional type: 4001
      torsional interaction through 2 and 1 with bead 6, torsional type: 4001

   bead 4: bead type 1002 [_CH], charge leader 4
      bonded to bead 3, type 2001, bond length:  1.40000, k/2:   236011.450
      bonded to bead 5, type 2001, bond length:  1.40000, k/2:   236011.450
      bending interaction through 5 with bead 6, bending type: 3001, bending angle:  120.000, k/2:   31702.5000
      bending interaction through 3 with bead 2, bending type: 3001, bending angle:  120.000, k/2:   31702.5000
      torsional interaction through 5 and 6 with bead 1, torsional type: 4001
      torsional interaction through 3 and 2 with bead 1, torsional type: 4001

   bead 5: bead type 1002 [_CH], charge leader 5
      bonded to bead 4, type 2001, bond length:  1.40000, k/2:   236011.450
      bonded to bead 6, type 2001, bond length:  1.40000, k/2:   236011.450
      bending interaction through 6 with bead 1, bending type: 3001, bending angle:  120.000, k/2:   31702.5000
      bending interaction through 4 with bead 3, bending type: 3001, bending angle:  120.000, k/2:   31702.5000
      torsional interaction through 6 and 1 with bead 2, torsional type: 4001
      torsional interaction through 4 and 3 with bead 2, torsional type: 4001

   bead 6: bead type 1002 [_CH], charge leader 6
      bonded to bead 5, type 2001, bond length:  1.40000, k/2:   236011.450
      bonded to bead 1, type 2001, bond length:  1.40000, k/2:   236011.450
      bending interaction through 1 with bead 2, bending type: 3001, bending angle:  120.000, k/2:   31702.5000
      bending interaction through 5 with bead 4, bending type: 3001, bending angle:  120.000, k/2:   31702.5000
      torsional interaction through 1 and 2 with bead 3, torsional type: 4001
      torsional interaction through 5 and 4 with bead 3, torsional type: 4001

MOLECULAR MASS:      78.00000

SPECIAL INTERACTION RULES
------------------------------------------
  Total: 0 exclusion rules for intermolecular interactions

  Total: 0 inclusion rules for intramolecular interactions

  Total: 0 special rules for intramolecular 1-5 OH interactions

SECTION UNIFORM_BIASING_POTENTIALS
------------------------------------------
Molecule type, biasing potential 1 through nbox [K]:
     0.000

SECTION SPECIFIC_ATOM_TRANSL
------------------------------------------
 natomtrans_atoms:              6
 atomtrans_atomlst:             1           1           1           1           1           1
 atomtrans_moleclst:            1           1           1           1           1           1

Molecule type: 1

LJ INCLUSION TABLE
       1   2   3   4   5   6
   1   F   F   F   F   F   F
   2   F   F   F   F   F   F
   3   F   F   F   F   F   F
   4   F   F   F   F   F   F
   5   F   F   F   F   F   F
   6   F   F   F   F   F   F

CHARGE INCLUSION TABLE
       1   2   3   4   5   6
   1   F   F   F   F   F   F
   2   F   F   F   F   F   F
   3   F   F   F   F   F   F
   4   F   F   F   F   F   F
   5   F   F   F   F   F   F
   6   F   F   F   F   F   F

1-4 LJ SCALING FACTORS
             1      2      3      4      5      6
      1  1.000  1.000  1.000  1.000  1.000  1.000
      2  1.000  1.000  1.000  1.000  1.000  1.000
      3  1.000  1.000  1.000  1.000  1.000  1.000
      4  1.000  1.000  1.000  1.000  1.000  1.000
      5  1.000  1.000  1.000  1.000  1.000  1.000
      6  1.000  1.000  1.000  1.000  1.000  1.000

1-4 CHARGE SCALING FACTORS
             1      2      3      4      5      6
      1  1.000  1.000  1.000  1.000  1.000  1.000
      2  1.000  1.000  1.000  1.000  1.000  1.000
      3  1.000  1.000  1.000  1.000  1.000  1.000
      4  1.000  1.000  1.000  1.000  1.000  1.000
      5  1.000  1.000  1.000  1.000  1.000  1.000
      6  1.000  1.000  1.000  1.000  1.000  1.000

PAIRWISE LJ AND COULOMB INTERACTIONS
    i    j         q0(i)         q0(j)     vvdW_1     vvdW_2 ...
 1002 1002      0.000000      0.000000   202.00       3.6950

NAMELIST TRANSFER
------------------------------------------
lopt_bias:  F
freq_opt_bias: 500

NAMELIST MC_VOLUME
------------------------------------------
target volume acceptance ratio (tavol): 0.50
frequency to adjust maximum volume displacement: 500
initial maximum volume displacement (rmvol): 1000.000
pmvol:   0.00000000
   pmvlmt for box 1:  1.000000000
nvolb: 1
cutoff will be addressed with option:  -1
   box pair 1: between 1 and 2,   pmvolb =  1.000000000

NAMELIST MC_SWATCH
------------------------------------------
pmswat:   0.00000000
   number of swatch pairs (nswaty): 1
   probability of each swatch pair:  1.000000000

   swatch molecule type pairs:    1    2
   nsampos: 1, ncut:  1  1, ncutsafe:  0  0
   molecule 1
   ncut 1: grow from 1, prev 0
   molecule 2
   ncut 1: grow from 1, prev 0
   splist:    1    1
   number of swatch box pairs: 1
   probability of the swatch box pair:  1.000000000
   box pair:    1    2

NAMELIST MC_SWAP
------------------------------------------
pmswap:   0.00000000
   swap probability for molecule type 1 (pmswmt):   1.0000
   number of swap box pairs for molecule type 1: 1
   pmswapb:  1.000000000
   box pair:    1    2

NAMELIST MC_CBMC
------------------------------------------
CBMC inner cutoff (rcutin):  5.000 [Ang]
AVBMC outer cutoff (rbsmax):  3.500 [Ang], inner cutoff (rbsmin):  2.500 [Ang]
lpresim:  F
iupdatefix: 10
pmcb:   0.00000000

molecule type: nchoi1  nchoi nchoir nchoih nchtor nchbna nchbnb icbdir icbsta first_bead_to_swap
            1:     12     10     50      1     50   1000   1000      0      0      1

molecule type:    pmcbmt         pmall  avbmc_version    pmbias        pmbsmt       pmbias2         pmfix   lrig
            1:  1.000000       0.00000              0   0.00000       0.00000       0.00000       0.00000      F

NAMELIST MC_FLUCQ
------------------------------------------
target fluctuating charge acceptance ratio (taflcq): 0.95
fluctuating charge temperature:    5.000 [K]
initial maximum displacement for fluctuating charge moves:  0.100000000E+00
pmflcq:   0.00000000
   nchoiq for each box:   1
nswapq: 0

molecule type:  lflucq lqtrans   pmfqmt            fqegp
            1:       F       F   1.0000           0.0000

NAMELIST MC_EE
------------------------------------------
pmexpc:   0.00000000
   expanded ens. prob. for molecule type 1 (pmeemt):   1.0000, lexpand:  F
pmexpc1:   0.00000000

NAMELIST MC_SIMPLE
------------------------------------------
pm_atom_tra:  0.500000000E-01
initial maximum displacement for atom translation:  0.100000000E-01
target translation acceptance ratio: 0.50
initial maximum displacement for molecule translation:  0.300000000
pmtra:  0.495000000
   translation probability for molecule type 1 (pmtrmt):   1.0000
target rotation acceptance ratio: 0.50
initial maximum displacement for molecule rotation:  0.400000000
pmrot:  1.000000000
   rotational probability for molecule type 1 (pmromt):   1.0000

percentage move probabilities:
 volume move       :    0.00 %
 swatch move       :    0.00 %
 swap move         :    0.00 %
 CBMC move         :    0.00 %
 fluct. charge move:    0.00 %
 expanded ens. move:    0.00 %
 new EE move       :    0.00 %
 atom trans. move  :    5.00 %
 translation move  :   44.50 %
 rotation move     :   50.50 %

NAMELIST EXTERNAL_FIELD
------------------------------------------
Steele 10-4-3 slit pore
On both sides of the simulation box (z = 0 & z = boxlz)
Surface material bead type: 190
Surface atom density:  0.11400 [Ang^-3]
Surface layer spacing:  3.40000 [Ang]
a1:  2.46000 [Ang]
Electric field in z direction:     0.00000000     [V/A]

READING CONFIGURATION FROM RESTART FILE
------------------------------------------
new maximum displacements read from restart-file
   max atom trans. displacement:     0.010000    0.010000    0.010000
   max volume displacement:   0.1000E+04

box      #1
   molecule type 1
      max trans. displacement:     0.300000   0.300000   0.300000
      max rot. displacement:       0.100000   0.100000   0.100000
      max fluc. q displacement:    0.100000

reading new box size from restart-file
 dimension box 1: a =    42.170000  b =    42.170000  c =    42.170000 rcut =   14.000000

Finished writing simulation box related info

 sumup control
 number of chains         400
number of chains of type    1 L1             400
 inter lj energy    48397108.5716805
 intra lj energy   0.000000000000000E+000
 bond vibration    3.266920382029358E-014
 bond bending      2.671487280527065E-005
 torsional         5.382616109272931E-006
 external          0.000000000000000E+000
 coulombic energy  0.000000000000000E+000
 fluc Q energy     0.000000000000000E+000
 well energy       0.000000000000000E+000
 total energy       48397108.5717126

 box             1  initial v   =    48397108.5717126
 surf. tension :   box 1 =     167.67124
 pressure check:   box 1 =   58390439.37
 compress factor:  box 1 =     792.88342

 +++++ start of markov chain +++++

 Cycle   Total   Energy    Boxlength    Pressure     Compress    Molecules
    50      50  0.4338E+07    42.170  58390439.4        0.00         400
   100     100  0.3849E+06    42.170  58390439.4        0.00         400
   150     150 -0.2979E+06    42.170  58390439.4        0.00         400
   200     200 -0.5670E+06    42.170  58390439.4        0.00         400
   250     250 -0.6814E+06    42.170  58390439.4        0.00         400
   300     300 -0.7123E+06    42.170  58390439.4        0.00         400
   350     350 -0.7512E+06    42.170  58390439.4        0.00         400
   400     400 -0.7849E+06    42.170  58390439.4        0.00         400
   450     450 -0.7963E+06    42.170  58390439.4        0.00         400
 Box            1
 Type  1 bn  29676.  29743.  29796.  33863.  33697.  33199.
         bs  21618.  21974.  21057.  29887.  29686.  29785.
 max.displ.       0.4348      0.4348      0.4348      0.1774      0.1774      0.1774
   500     500 -0.7742E+06    42.170    398528.9        5.41         400
   550     550 -0.7580E+06    42.170    398528.9        5.41         400
   600     600 -0.7516E+06    42.170    398528.9        5.41         400
   650     650 -0.7343E+06    42.170    398528.9        5.41         400
   700     700 -0.7152E+06    42.170    398528.9        5.41         400
   750     750 -0.7114E+06    42.170    398528.9        5.41         400
   800     800 -0.6767E+06    42.170    398528.9        5.41         400
   850     850 -0.6937E+06    42.170    398528.9        5.41         400
   900     900 -0.6530E+06    42.170    398528.9        5.41         400
   950     950 -0.6479E+06    42.170    398528.9        5.41         400
 Box            1
 Type  1 bn  29477.  29822.  29618.  33792.  33601.  33577.
         bs  22238.  22509.  22155.  30754.  30589.  30698.
 max.displ.       0.6543      0.6543      0.6543      0.3234      0.3234      0.3234
  1000    1000 -0.6593E+06    42.170    248043.4        3.37         400
  1050    1050 -0.6099E+06    42.170    248043.4        3.37         400
  1100    1100 -0.5669E+06    42.170    248043.4        3.37         400
  1150    1150 -0.5495E+06    42.170    248043.4        3.37         400
  1200    1200 -0.5082E+06    42.170    248043.4        3.37         400
  1250    1250 -0.5135E+06    42.170    248043.4        3.37         400
  1300    1300 -0.4696E+06    42.170    248043.4        3.37         400
  1350    1350 -0.4860E+06    42.170    248043.4        3.37         400
  1400    1400 -0.4482E+06    42.170    248043.4        3.37         400
  1450    1450 -0.4417E+06    42.170    248043.4        3.37         400
 Box            1
 Type  1 bn  30109.  29417.  29540.  33607.  33519.  33648.
         bs  20378.  20048.  19691.  28647.  28582.  28887.
 max.displ.       0.8832      0.8832      0.8832      0.5527      0.5527      0.5527
  1500    1500 -0.3892E+06    42.170    301502.4        4.09         400
  1550    1550 -0.4068E+06    42.170    301502.4        4.09         400
  1600    1600 -0.4030E+06    42.170    301502.4        4.09         400
  1650    1650 -0.3892E+06    42.170    301502.4        4.09         400
  1700    1700 -0.3732E+06    42.170    301502.4        4.09         400
  1750    1750 -0.3730E+06    42.170    301502.4        4.09         400
  1800    1800 -0.3497E+06    42.170    301502.4        4.09         400
  1850    1850 -0.3865E+06    42.170    301502.4        4.09         400
  1900    1900 -0.3553E+06    42.170    301502.4        4.09         400
  1950    1950 -0.3255E+06    42.170    301502.4        4.09         400
 Box            1
 Type  1 bn  29727.  29687.  29501.  33638.  33612.  33813.
         bs  17929.  17910.  17659.  26019.  25950.  26372.
 max.displ.       1.0628      1.0628      1.0628      0.8569      0.8569      0.8569
  2000    2000 -0.3201E+06    42.170    221397.9        3.01         400
  2050    2050 -0.3173E+06    42.170    221397.9        3.01         400
  2100    2100 -0.3178E+06    42.170    221397.9        3.01         400
  2150    2150 -0.2818E+06    42.170    221397.9        3.01         400
  2200    2200 -0.2911E+06    42.170    221397.9        3.01         400
  2250    2250 -0.2638E+06    42.170    221397.9        3.01         400
  2300    2300 -0.2623E+06    42.170    221397.9        3.01         400
  2350    2350 -0.2952E+06    42.170    221397.9        3.01         400
  2400    2400 -0.2986E+06    42.170    221397.9        3.01         400
  2450    2450 -0.2876E+06    42.170    221397.9        3.01         400
 Box            1
 Type  1 bn  29549.  29757.  29703.  33684.  33559.  33884.
         bs  16085.  16221.  16119.  22900.  22911.  23348.
 max.displ.       1.1565      1.1565      1.1565      1.1720      1.1720      1.1720
  2500    2500 -0.2867E+06    42.170    216419.6        2.94         400
  2550    2550 -0.2602E+06    42.170    216419.6        2.94         400
  2600    2600 -0.2371E+06    42.170    216419.6        2.94         400
  2650    2650 -0.2692E+06    42.170    216419.6        2.94         400
  2700    2700 -0.2385E+06    42.170    216419.6        2.94         400
  2750    2750 -0.2209E+06    42.170    216419.6        2.94         400
  2800    2800 -0.2470E+06    42.170    216419.6        2.94         400
  2850    2850 -0.2004E+06    42.170    216419.6        2.94         400
  2900    2900 -0.2282E+06    42.170    216419.6        2.94         400
  2950    2950 -0.2175E+06    42.170    216419.6        2.94         400
 Box            1
 Type  1 bn  29946.  29802.  29922.  33373.  33393.  33570.
         bs  15563.  15508.  15083.  20244.  20095.  20616.
 max.displ.       1.1905      1.1905      1.1905      1.4239      1.4239      1.4239
  3000    3000 -0.2862E+06    42.170    197450.2        2.68         400
  3050    3050 -0.2671E+06    42.170    197450.2        2.68         400
  3100    3100 -0.2263E+06    42.170    197450.2        2.68         400
  3150    3150 -0.2018E+06    42.170    197450.2        2.68         400
  3200    3200 -0.2226E+06    42.170    197450.2        2.68         400
  3250    3250 -0.2200E+06    42.170    197450.2        2.68         400
  3300    3300 -0.2582E+06    42.170    197450.2        2.68         400
  3350    3350 -0.2178E+06    42.170    197450.2        2.68         400
  3400    3400 -0.1956E+06    42.170    197450.2        2.68         400
  3450    3450 -0.2108E+06    42.170    197450.2        2.68         400
 Box            1
 Type  1 bn  29496.  29431.  29706.  33964.  33643.  33885.
         bs  15126.  15068.  15102.  19186.  18874.  18967.
 max.displ.       1.2168      1.2168      1.2168      1.6002      1.6002      1.6002
  3500    3500 -0.1948E+06    42.170    219487.9        2.98         400
  3550    3550 -0.1371E+06    42.170    219487.9        2.98         400
  3600    3600 -0.1579E+06    42.170    219487.9        2.98         400
  3650    3650 -0.1972E+06    42.170    219487.9        2.98         400
  3700    3700 -0.1953E+06    42.170    219487.9        2.98         400
  3750    3750 -0.1963E+06    42.170    219487.9        2.98         400
  3800    3800 -0.2032E+06    42.170    219487.9        2.98         400
  3850    3850 -0.1991E+06    42.170    219487.9        2.98         400
  3900    3900 -0.1748E+06    42.170    219487.9        2.98         400
  3950    3950 -0.1427E+06    42.170    219487.9        2.98         400
 Box            1
 Type  1 bn  29434.  29688.  29650.  33713.  33899.  33566.
         bs  14601.  14891.  15016.  18267.  18093.  17664.
 max.displ.       1.2201      1.2201      1.2201      1.7088      1.7088      1.7088
  4000    4000 -0.1852E+06    42.170    180157.8        2.45         400
  4050    4050 -0.1792E+06    42.170    180157.8        2.45         400
  4100    4100 -0.1875E+06    42.170    180157.8        2.45         400
  4150    4150 -0.1704E+06    42.170    180157.8        2.45         400
  4200    4200 -0.2079E+06    42.170    180157.8        2.45         400
  4250    4250 -0.1573E+06    42.170    180157.8        2.45         400
  4300    4300 -0.1500E+06    42.170    180157.8        2.45         400
  4350    4350 -0.1758E+06    42.170    180157.8        2.45         400
  4400    4400 -0.1876E+06    42.170    180157.8        2.45         400
  4450    4450 -0.1662E+06    42.170    180157.8        2.45         400
 Box            1
 Type  1 bn  29542.  29727.  29936.  33948.  33172.  33726.
         bs  14561.  14467.  15111.  17287.  17210.  17123.
 max.displ.       1.2074      1.2074      1.2074      1.7495      1.7495      1.7495
  4500    4500 -0.1940E+06    42.170    204500.1        2.78         400
  4550    4550 -0.1935E+06    42.170    204500.1        2.78         400
  4600    4600 -0.1874E+06    42.170    204500.1        2.78         400
  4650    4650 -0.1950E+06    42.170    204500.1        2.78         400
  4700    4700 -0.2381E+06    42.170    204500.1        2.78         400
  4750    4750 -0.2358E+06    42.170    204500.1        2.78         400
  4800    4800 -0.2048E+06    42.170    204500.1        2.78         400
  4850    4850 -0.2333E+06    42.170    204500.1        2.78         400
  4900    4900 -0.2395E+06    42.170    204500.1        2.78         400
  4950    4950 -0.2711E+06    42.170    204500.1        2.78         400
 Box            1
 Type  1 bn  29743.  29682.  29657.  33736.  33657.  33545.
         bs  14839.  14659.  14872.  17299.  17477.  17143.
 max.displ.       1.2028      1.2028      1.2028      1.7998      1.7998      1.7998
  5000    5000 -0.2627E+06    42.170    196367.2        2.67         400

 +++++ end of markov chain +++++

 ### Atom Translations ###

 x-dir: attempts =   33416.0   ratio = 0.534   max.displ. = 0.6747E+00
 x-dir: attempts =   33241.0   ratio = 0.534   max.displ. = 0.6684E+00
 x-dir: attempts =   33376.0   ratio = 0.536   max.displ. = 0.6756E+00

 ### Translations ###

molecule typ = 1   L1          in box           1
 x-dir: attempts =  296699.0   ratio = 0.583   max.displ. = 0.1203E+01
 y-dir: attempts =  296756.0   ratio = 0.584   max.displ. = 0.1203E+01
 z-dir: attempts =  297029.0   ratio = 0.579   max.displ. = 0.1203E+01

 ### Rotations ###

molecule typ = 1   L1          in box           1
 x-dir: attempts =  337318.0   ratio = 0.683   max.displ. = 0.1800E+01
 y-dir: attempts =  335752.0   ratio = 0.683   max.displ. = 0.1800E+01
 z-dir: attempts =  336413.0   ratio = 0.685   max.displ. = 0.1800E+01


 ### Volume change       ###
 attempts =     0.0   ratio = 0.000max.displ. = 0.1000E+04   avg.acc.displ. = 0.0000E+00

### Molecule swap       ###

molecule typ = 1   L1
between box 1 and 2 into box 1   uattempts = 0 attempts = 0   accepted = 0
between box 1 and 2 into box 2   uattempts = 0 attempts = 0   accepted = 0
number of times move in: 0  accepted = 0
number of times move out: 0  accepted = 0

 ### Charge Fluctuation  ###


 ### Expanded Ensemble Move  ###


New Biasing Potential

molecule  1:   0.00000000

 sumup control
 number of chains         400
number of chains of type    1 L1             400
 inter lj energy   -855250.150715562
 intra lj energy   0.000000000000000E+000
 bond vibration     314096.291184030
 bond bending       90352.7927093912
 torsional          188091.895388632
 external          0.000000000000000E+000
 coulombic energy  0.000000000000000E+000
 fluc Q energy     0.000000000000000E+000
 well energy       0.000000000000000E+000
 total energy      -262709.171433509
 ### problem with total energy ###  box            1
 summup, Markov chain, abs_err, rel_err:
  -262709.171433509       -262709.171435301       1.792330294847488E-006  6.822488476725658E-012
 ### problem with inter mol.en.  ###
 summup, Markov chain, abs_err, rel_err:
  -855250.150715562       -855250.150717300       1.738662831485271E-006  2.032928997471734E-012

 vstart       =     48397108.5717126131
 vend         =      -262709.1714335090
 vbox         =      -262709.1714353013

 Averages and fluctuations                                  Box  1

 pressure                                      [kPa] =    238385.53
 pressure                         [simulation units] =    17.266196
 surface tension                              [mN/m] =      16.9825
 compress factor                                     =      3.23704
 chem. potential of type   1 L1                  [K] =        0.000
 boxlength                                       [A] =       42.170
 boxlength                                       [A] =       42.170
 boxlength                                       [A] =       42.170
 no. of chains of type     1 L1                      =      400.000
 molar volume                             [cm^3/mol] =      112.902
 specific density                           [g/cm^3] =     0.690864
 number density of type    1 L1         [chain/nm^3] =      5.33395
 molfraction of type       1 L1                      =    1.0000000
 mean sete length of type  1 L1                [A^2] =        1.964

 Total energy [K per system and kJ/mol per chain] =    -162199.12      -3.37
 Inter LJ     [K per system and kJ/mol per chain] =    -616510.56     -12.81
 Tail  LJ     [K per system and kJ/mol per chain] =          0.00       0.00
 Intra LJ     [K per system and kJ/mol per chain] =          0.00       0.00
 Stretch      [K per system and kJ/mol per chain] =     261550.11       5.44
 Bond bending [K per system and kJ/mol per chain] =      70222.87       1.46
 Torsion      [K per system and kJ/mol per chain] =     122538.46       2.55
 Coulomb      [K per system and kJ/mol per chain] =          0.00       0.00
 External pot [K per system and kJ/mol per chain] =          0.00       0.00
 3-body Garo  [K per system and kJ/mol per chain] =          0.00       0.00
 Fluc Q       [K per system and kJ/mol per chain] =          0.00       0.00

 fluctuation in <vtot> =  2022744.47

 type  box       vinter       vintra         vtor        vbend        vtail
    1    1      0.00000      0.00000      0.00000      0.00000      0.00000


 specific density box  1 =  0.69086E+00  0.40809E-10  0.20405E-10
 system volume    box  1 =  0.74991E+05  0.16060E-05  0.80302E-06
 pressure         box  1 =  0.23839E+06   48284.       24142.
 surface tension  box  1 =     16.98253     62.89627     31.44814
 compressibility  box  1 =      3.23704      0.65565      0.32782

 Total energy     box  1 = -0.16220E+06  0.24818E+06  0.12409E+06
 Inter LJ         box  1 = -0.61651E+06  0.41790E+06  0.20895E+06
 Tail  LJ         box  1 =  0.00000E+00  0.00000E+00  0.00000E+00
 Intra LJ         box  1 =  0.00000E+00  0.00000E+00  0.00000E+00
 Stretch          box  1 =  0.26155E+06  0.10059E+06  0.50297E+05
 Bond bending     box  1 =  0.70223E+05  0.31002E+05  0.15501E+05
 Torsion          box  1 =  0.12254E+06  0.82184E+05  0.41092E+05
 Coulomb          box  1 =  0.00000E+00  0.00000E+00  0.00000E+00
 External pot     box  1 =  0.00000E+00  0.00000E+00  0.00000E+00
 3-body Garo      box  1 =  0.00000E+00  0.00000E+00  0.00000E+00
 Fluc Q           box  1 =  0.00000E+00  0.00000E+00  0.00000E+00

 Enthalpy Inst.[kJ/mol] for box  1 =      19.5301       1.9595       0.9798
 Enthalpy Ext. [kJ/mol] for box  1 =      -7.1289       3.9309       1.9654

 Cp residual(J/Kmol) =   85046.4539
   H2 =  0.411397229E+13
   H  =  -149923.712

 chemical potential  itype  1 box  1 =         0.000        0.000        0.000
 mean sete length    itype  1 box  1 =         1.964        0.003        0.001
 number density      itype  1 box  1 =   0.53340E+01  0.31507E-09  0.15754E-09
 mole fraction       itype  1 box  1 =     1.0000000    0.0000000    0.0000000


 -----block averages ------
  ------------ box:  1
 block    energy     density    pressure       Z        surf ten   mol fracs
   1  0.3014E+06  0.6909E+00  0.3233E+06  0.4390E+01  0.7718E+02  0.1000E+01
   2 -0.4429E+06  0.6909E+00  0.2615E+06  0.3550E+01 -0.8707E+02  0.1000E+01
   3 -0.2684E+06  0.6909E+00  0.2069E+06  0.2810E+01  0.6810E+02  0.1000E+01
   4 -0.2051E+06  0.6909E+00  0.1998E+06  0.2713E+01 -0.2332E+02  0.1000E+01
   5 -0.1960E+06  0.6909E+00  0.2004E+06  0.2722E+01  0.5003E+02  0.1000E+01
 Program ended at 2021-09-14 19:12:49.627
`

Reply:
Ah this makes sense then. Yeah this a pretty good use of grep and awk.